### PR TITLE
Add OIDC Scope REST API

### PR DESCRIPTION
**Pull request by sarubi**
## Purpose
Related to wso2/product-is#7130
Depends on wso2-extensions/identity-inbound-auth-oauth#1302

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.common/pom.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            <artifactId>spring-web</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.inbound.auth.oauth2</groupId>
+            <artifactId>org.wso2.carbon.identity.oauth</artifactId>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+</project>
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.common/pom.xml]: Add new line

File: components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.common/pom.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.common/pom.xml]: License year 2020.

File: components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.common/src/main/java/org/wso2/carbon/identity/api/server/oidc/scope/management/common/OidcScopeConstants.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    public static final String OIDC_SCOPE_API_PATH_COMPONENT = "/oidc/scopes";
+    public static final String PATH_SEPERATOR = "/";
+
+    /**
+     * Enum for OIDC scope management service related errors.
+     */
+    public enum ErrorMessage {
+
+        INVALID_REQUEST("60001", "Invalid Request."),
+        ERROR_CONFLICT_REQUEST("41004", "Scope is already exists.");
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.common/src/main/java/org/wso2/carbon/identity/api/server/oidc/scope/management/common/OidcScopeConstants.java]:
```suggestion
        ERROR_CONFLICT_REQUEST("41004", "Scope already exists.");
```

File: components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.common/src/main/java/org/wso2/carbon/identity/api/server/oidc/scope/management/common/OidcScopeConstants.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+package org.wso2.carbon.identity.api.server.oidc.scope.management.common;
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.common/src/main/java/org/wso2/carbon/identity/api/server/oidc/scope/management/common/OidcScopeConstants.java]: License

File: components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.v1/src/main/java/org/wso2/carbon/identity/api/server/oidc/scope/management/v1/core/OidcScopeManagementService.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
+package org.wso2.carbon.identity.api.server.oidc.scope.management.v1.core;
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.v1/src/main/java/org/wso2/carbon/identity/api/server/oidc/scope/management/v1/core/OidcScopeManagementService.java]: License

File: components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.v1/src/main/java/org/wso2/carbon/identity/api/server/oidc/scope/management/v1/core/OidcScopeManagementService.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+     */
+    private List<Scope> buildScopeList(ScopeDTO[] scopeDTOS) {
+
+        List<Scope> scopeList = new ArrayList<>();
+
+        for (int i = 0; i < scopeDTOS.length; i++) {
+            scopeList.add(convertScopeDTOObjectToScope(scopeDTOS[i]));
+        }
+        return scopeList;
+
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.v1/src/main/java/org/wso2/carbon/identity/api/server/oidc/scope/management/v1/core/OidcScopeManagementService.java]: Unnecessary new line

File: components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.v1/src/main/java/org/wso2/carbon/identity/api/server/oidc/scope/management/v1/core/OidcScopeManagementService.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+     */
+    private Scope convertScopeDTOObjectToScope(ScopeDTO scopeDTO) {
+
+        Scope scope = new Scope();
+        scope.setName(scopeDTO.getName());
+        scope.setDisplayName(scopeDTO.getDisplayName());
+        scope.setDescription(scopeDTO.getDescription());
+        scope.setClaims(Arrays.asList(scopeDTO.getClaim()));
+        return scope;
+
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.oidc.scope.management/org.wso2.carbon.identity.api.server.oidc.scope.management.v1/src/main/java/org/wso2/carbon/identity/api/server/oidc/scope/management/v1/core/OidcScopeManagementService.java]: Unnecessary new line

**Comment by sarubi** [comment]
@madurangasiriwardena Fixed the requested changes

**Pull request merged**